shared/src/ledger/native_vp.rs

```diff
@@ -241,3 +268,78 @@ where
         }
     }
 }
+
+impl<'a, DB, H> ClientReader for Ctx<'a, DB, H>
+where
+    DB: 'static + storage::DB + for<'iter> storage::DBIter<'iter>,
+    H: 'static + StorageHasher,
+{
+    fn client_type(&self, client_id: &ClientId) -> Option<ClientType> {
+        let path = Path::ClientType(client_id.clone()).to_string();
+        let key = Key::ibc_key(path)
+            .expect("Creating a key for a client type shouldn't fail");
+        match self.read_post(&key) {
+            Ok(Some(value)) => {
+                let s: String = storage::types::decode(&value)
+                    .expect("decoding shouldn't fail");
```

Comment:
A malicious tx could write a value that would fail decoding here and panic. We can use `.ok()` on the `Result` instead. `Option` type also supports the `?` syntax, so this could be:
```suggestion
                    .ok()?;
```

Same for all the `expect`s below

Reply:
That makes sense! Thanks!